packages/cdk8s/lib/app.ts

```diff
@@ -32,12 +36,61 @@ export class App extends Construct {
    * Synthesizes all manifests to the output directory
    */
   public synth(): void {
-    if (!fs.existsSync(this.outdir)) {
-      fs.mkdirSync(this.outdir);
+
+    fs.mkdirSync(this.outdir, { recursive: true });
+
+    const hasChartDependencies = this.inferChartDependencies();
```

Comment:
Why can’t this be called inside produceManifest?

Reply:
`produceManifests` is called after the validation part and I think the validation should be aware of the inferred chart dependencies. Originally this was just one big method, but I like this separation, even though I agree returning a `boolean` and then accepting it is a bit awkward.